code/modules/mob/mob_movement.dm

```diff
@@ -553,18 +553,10 @@
 		var/atom/loc_atom = loc
 		return loc_atom.relaymove(src, UP)
 
-	var/ventcrawling_flag = HAS_TRAIT(src, TRAIT_MOVE_VENTCRAWLING) ? ZMOVE_VENTCRAWLING : 0
-
-	if(can_z_move(DOWN, above_turf, current_turf, ZMOVE_FALL_FLAGS|ventcrawling_flag)) //Will we fall down if we go up?
-		if(buckled)
-			to_chat(src, span_warning("[buckled] is is not capable of flight."))
-		else
-			to_chat(src, span_warning("You are not Superman."))
-		return
 	balloon_alert(src, "moving up...")
 	if(!do_after(src, 1 SECONDS, hidden = TRUE))
```

Comment:
This do_after shouldn't be running if you can't move up before it, though.

Reply:
Would it be better if it was like this? I am not against it, I just made it this way since the majority of cases was waiting 1 second before getting feedback on the action and it felt out of place that is only specific cases it was not like this. I could probably implement it so the check happens before.

Reply:
I think it's (generally) more annoying to have to wait a second to be told you can't do something instead of being told outright. 

Since the proc itself checks if you can move, think of it as safety:
`check if we can move` -> `if so, wait 1 second` -> `if we can still do the move, do the move`.